sdk/storage/azure-storage-files-shares/src/share_directory_client.cpp

```diff
@@ -590,6 +590,27 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       directoryHandle.OpenedOn = std::move(handle.OpenedOn);
       directoryHandle.ParentId = std::move(handle.ParentId);
       directoryHandle.SessionId = std::move(handle.SessionId);
+      if (!handle.AccessRightList.empty())
+      {
+        Models::ShareFileHandleAccessRights accessRights
+            = static_cast<Models::ShareFileHandleAccessRights>(0);
+        for (auto& accessRight : handle.AccessRightList)
+        {
+          if (accessRight == Models::AccessRight::Read)
```

Comment:
should we move `Models::AccessRight` to `_detail` namespace?

Reply:
Yeah, I have moved them to `_detail` namemspace.